src/Health/Interfaces/HealthInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+declare(strict_types=1);
+
+namespace EoneoPay\Externals\Health\Interfaces;
+
+interface HealthInterface
+{
+    /**
+     * Represents a degraded state.
+     *
+     * @const int
+     */
+    public const STATE_DEGRADED = 0;
+
+    /**
+     * Represents a healthy state.
+     *
+     * @const int
+     */
+    public const STATE_HEALTHY = 100;
+
+    /**
+     * Performs an extended health check.
+     *
+     * @return mixed[]
+     */
+    public function extended(): array;
```

Comment:
Wouldn't it be better for this tor return an interface or DTO with the check & message embedded in it?

Reply:
it couldn't hurt